### PR TITLE
627 Product Family detail page - Authored content UI

### LINKED ISSUE #619: [Product] Product/Product Family detail page - part 3 (content)
- [ ] Product specification attributes
- [ ] Authored content on each tab
- [ ] Citations embeds

### PR DESCRIPTION
**Pull request by sharanagoudapatil**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #619  

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/
- After: https://619-product-spec-attributes-ui--danaher-ls-aem--hlxsites.hlx.live/us/en/products/family/mica
- After: https://619-product-spec-attributes-ui--danaher-ls-aem--hlxsites.hlx.live/us/en/products/family/spectramax-id3-id5-series-microplate-readers
- After: https://619-product-spec-attributes-ui--danaher-ls-aem--hlxsites.hlx.live/us/en/products/family/cloneselect-imagers
- After: https://619-product-spec-attributes-ui--danaher-ls-aem--hlxsites.hlx.live/us/en/products/family/clonepix-2-mammalian-colony-pickers

### CODE REVIEW

File: styles/tailwind.css [hunk 1]
```diff
@@ -10,6 +10,7 @@
  * governing permissions and limitations under the License.
  */
 
+@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap');
```
Comment (mhaack) [inline styles/tailwind.css:13]: We must not load the fonts from google directly. Add these to the /fonts folder and load from there.

File: styles/fonts.css [hunk 2]
```diff
@@ -1,4 +1,5 @@
 @import url("https://use.typekit.net/oip4drb.css");
+@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap');
```
Comment (mhaack) [inline styles/fonts.css:2]: I think this is still wrong/not needed. We already the Inter font defined with different font weights, see below starting line 22. What is missing there?

Comment (mhaack) [reply styles/fonts.css:2]: I checked below you have Inter with font-weight 500, 600 & 800. 500 & 600 are defined, so we only need to add 800 below.

Comment (sharanagoudapatil) [reply styles/fonts.css]: Yes, font-weight 800 added now in the fonts.css with required 'woff2' file and changes looks fine now. Please have a look. Thanks.

**Review by davenichols-DHLS — APPROVED** [review]
The content appears to be showing and working as I can see.  I will let the others review.

File: blocks/columns/columns.css [hunk 3]
```diff
@@ -72,6 +72,52 @@ main .top-border > div.columns-wrapper:first-child {
...
+.productdetail main .columns p.embed-vimeo {
+  @apply my-2;
+}
+
+.productdetail main .columns p:has(picture)~p, 
+.productdetail main .columns p.embed-vimeo~p {
+  @apply text-sm leading-5 text-gray-300;
+}
+
+.productdetail main .columns div:has(picture)~div:has(h2) {
```
Comment (duynguyen) [inline blocks/columns/columns.css:86]: is this an important style? meaning does it break if `has` is not working correctly?
note that `has` is not supported in older Firefox versions which is still popular today.

File: blocks/columns/columns.css [hunk 4]
```diff
@@ -72,6 +72,52 @@ main .top-border > div.columns-wrapper:first-child {
...
+
+.productdetail main .default-content-wrapper > p {
+  @apply w-3/4;
+}
+
+.productdetail main .default-content-wrapper > h2:only-child {
+  @apply text-center;
+}
+
+.features-card-left .h-full p:has(span) + p,
```
Comment (duynguyen) [inline blocks/columns/columns.css:110]: same comment regarding `has`

File: blocks/columns/columns.css [hunk 5]
```diff
@@ -72,6 +72,52 @@ main .top-border > div.columns-wrapper:first-child {
...
+.cols-3 .h-full p:has(span) + p {
+  @apply text-lg font-medium text-gray-900 leading-5;
+}
+ 
+.features-card-left .h-full p:has(span) + p + p:not(span) {
```
Comment (duynguyen) [inline blocks/columns/columns.css:115]: same comment regarding `has`

File: blocks/columns/columns.js [hunk 6]
```diff
@@ -25,6 +25,31 @@ export default function decorate(block) {
...
       }
 
+      const ulEle = row.querySelectorAll('div > ul');
+      if (ulEle.length > 0) {
+        ulEle.forEach((ele) => {
+          ele.classList.add(...'text-base list-disc pl-10 space-y-2 text-danahergray-700'.split(' '));
+        });
+      }
+      const pElements = row.querySelectorAll('p');
+      if (pElements) {
```
Comment (duynguyen) [inline blocks/columns/columns.js:35]: do we need this condition here? pElements is a list, and if it is empty, the foreach loop won't run.

File: blocks/columns/columns.js [hunk 7]
```diff
@@ -25,6 +25,31 @@ export default function decorate(block) {
...
+        [...pElements].forEach((item) => {
+          const ulElements = item.parentElement.querySelectorAll('ul');
+          if (ulElements.length > 0) {
+            ulElements.forEach((element) => {
+              element.classList.add(...'text-base list-disc pl-10 space-y-2 text-danahergray-700'.split(' '));
+            });
+          }
+          const spanEl = item.querySelectorAll('p > span.icon');
+          if (spanEl.length > 0) {
```
Comment (duynguyen) [inline blocks/columns/columns.js:44]: same comment regarding redundant condition before a loop

**Comment by sharanagoudapatil** [comment]
Hi @duynguyen - I've addressed all above comments from you. Please have a look. Thanks.

File: blocks/columns/columns.js (hunk 7 above)
Comment (sharanagoudapatil) [reply blocks/columns/columns.js]: It is addressed. Thanks.

File: blocks/columns/columns.css (hunk 4 above)
Comment (sharanagoudapatil) [reply blocks/columns/columns.css]: Removed has from css. Thanks.

File: blocks/columns/columns.css [hunk 8]
```diff
@@ -72,6 +72,51 @@ main .top-border > div.columns-wrapper:first-child {
   @apply py-2 px-6 !text-white;
 }
 
+/* productdetail changes from here */
+
+.productdetail main .columns p.embed-vimeo {
+  @apply my-2;
+}
+
+.productdetail main .columns.columns-2-cols p:not(picture) + p:last-child,
```
Comment (duynguyen) [inline blocks/columns/columns.css:81]: does this break the paragraph styling here? https://619-product-spec-attributes-ui--danaher-ls-aem--hlxsites.hlx.page/us/en/products/family/spectramax-id3-id5-series-microplate-readers

Comment (sharanagoudapatil) [reply blocks/columns/columns.css:81]: Yes @duynguyen it's breaking here, as we know the pages under product family detail pages are not similar. Here we are not having anything to differentiate between these two p tags. I'll check with @rgravitvl if we add any attribute/class to this p tag to handle this CSS issue.

File: blocks/columns/columns.js [hunk 9]
```diff
@@ -25,6 +25,31 @@ export default function decorate(block) {
...
+      if (ulEle.length > 0) {
+        ulEle.forEach((ele) => {
+          ele.classList.add(...'text-base list-disc pl-10 space-y-2 text-danahergray-700'.split(' '));
+        });
+      }
+      const pElements = row.querySelectorAll('p');
+      if (pElements) {
+        [...pElements].forEach((item) => {
+          const ulElements = item.parentElement.querySelectorAll('ul');
+          if (ulElements.length > 0) {
```
[the remark this answers is not in the document]
Comment (sharanagoudapatil) [reply blocks/columns/columns.js]: Removed unnecessary conditions.

File: blocks/columns/columns.js [hunk 10]
```diff
@@ -25,6 +25,31 @@ export default function decorate(block) {
         row.classList.add('h-full');
       }
 
+      const ulEle = row.querySelectorAll('div > ul');
+      if (ulEle.length > 0) {
```
[the remark this answers is not in the document]
Comment (sharanagoudapatil) [reply blocks/columns/columns.js]: Removed it as not required.

File: styles/fonts.css (hunk 2 above)
Comment (sharanagoudapatil) [reply styles/fonts.css]: updated the fonts.

File: blocks/columns/columns.js [hunk 11]
```diff
@@ -25,6 +29,48 @@ export default function decorate(block) {
...
+            } else {
+              rightDiv.append(element);
+              cardDiv.append(rightDiv);
+            }
+          });
+          row.append(cardDiv);
+        }
+      } else if (block.className.includes('columns-2-cols')) {
+        const pTags = row.querySelectorAll('p');
+        if (pTags.length > 0) {
```
Comment (duynguyen) [inline blocks/columns/columns.js:65]: this condition could be removed as it's not needed

File: blocks/columns/columns.js [hunk 12]
```diff
@@ -25,6 +29,48 @@ export default function decorate(block) {
...
+      });
+
+      if (block.className.includes('features-card-left')) {
+        const pTags = row.querySelectorAll('p');
+        if (pTags.length > 0) {
+          let cardDiv;
+          let rightDiv;
+          pTags.forEach((element) => {
+            if (element.firstElementChild?.nodeName.toLowerCase() === 'span') {
+              if (cardDiv) row.append(cardDiv);
```
Comment (duynguyen) [inline blocks/columns/columns.js:51]: the logics here are a bit complex. why don't we initialize cardDiv and append it to row?

File: blocks/columns/columns.js [hunk 13]
```diff
@@ -25,6 +29,48 @@ export default function decorate(block) {
...
+              element.classList.add('left-content');
+              cardDiv = div({ class: 'card' });
+              cardDiv.append(element);
+              rightDiv = div({ class: 'right-content' });
+            } else {
+              rightDiv.append(element);
+              cardDiv.append(rightDiv);
```
Comment (duynguyen) [inline blocks/columns/columns.js:58]: what if cardDiv is not yet initialized (aka null) ? would recommend to refine the logics here.

File: blocks/columns/columns.js (hunk 11 above)
Comment (sharanagoudapatil) [reply blocks/columns/columns.js]: It is removed.

File: blocks/columns/columns.js (hunk 12 above)
Comment (sharanagoudapatil) [reply blocks/columns/columns.js]: As discussed, updated the logic bit. Please check. Thanks.

File: blocks/columns/columns.js (hunk 13 above)
Comment (sharanagoudapatil) [reply blocks/columns/columns.js]: As discussed, updated the logic bit. Please check. Thanks.

**Review by duynguyen — APPROVED** [review]

**Pull request merged**